File: CounterApp/app/src/main/java/com/example/counterapp/MainActivity.java
package com.example.counterapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.example.counterapp.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private int counter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setOnClickListener();

    }

    private void setOnClickListener () {
        binding.btnCounter.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                binding.tvCounter.setText(""+increaseCounter());
            }
        });

    }

    private int increaseCounter()
    {
        return ++counter;
    }
}

File: Dicoding/AndroidExpert/ExpertSubmissions/app/src/main/java/com/example/expertsubmissions/MainActivity.kt
package com.example.expertsubmissions

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.core.data.implementations.MovieRepositoryImpl
import com.example.expertsubmissions.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    private lateinit var appBarConfiguration : AppBarConfiguration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.appBarMain.toolbar)
        setupDrawer()

    //    val movieRepositoryImpl = MovieRepositoryImpl()
        /*binding.btn.setOnClickListener {
            val moveBro = Uri.parse("favorite_scheme://favorite_host")
            startActivity(Intent(Intent.ACTION_VIEW,moveBro))
        }

         */
    }
    private fun setupDrawer()
    {
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        appBarConfiguration = AppBarConfiguration(setOf(R.id.nav_home, R.id.nav_detail), binding.drawerLayout)
        setupActionBarWithNavController(navController, appBarConfiguration)
        binding.navView.setupWithNavController(navController)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.activity_main_drawer,menu)
        return true
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }

}